frontend/src/metabase/query_builder/components/filters/pickers/TextPicker.jsx

```diff
@@ -67,7 +68,16 @@ export default class TextPicker extends Component {
 
     return (
       <div>
-        <div className="FilterInput px1 pt1 relative">
+        <div className="FilterInput px1 pt1 relative flex align-center">
+          {!!prefix && (
+            <span
+              data-testid="input-prefix"
+              className="text-medium px1"
+              style={{ marginRight: -30, width: 30, zIndex: 2 }}
```

Comment:
Yeah, I can feel your pain here. This component is messy and it would require a lot more effort to implement it differently